music_server.py

```diff
@@ -14,6 +16,42 @@ def page_not_found(e):
 	print(e)
 	return '404 Not Found', 404
 
+@app.route('/albumsandtracks')
+def get_albums_with_tracks():
+	"""
+	Retrieve a complete list of albums and the contents
+	of each album on the server.
+	"""
+
+	# Initialize an empty dictionary to store albums and their tracks
+	albums={}
+
+	# Relative path to the Music directory with Albums
+	albums_path = 'Music'
+
+	# Loop through the albums
+	for album in os.listdir(albums_path):
+
+		# Construct the path to the current album
+		album_path = os.path.join(albums_path, album)
+
+		# Check if the directory specified by album_path exists
+		if os.path.isdir(album_path):
+
+			# Initialize an empty list of tracks
+			tracks = []
+			# List all tracks in the current album directory
+			for track in os.listdir(album_path):
+				if os.path.isfile(os.path.join(album_path, track)):
+					tracks.append(track)
+
+			# Add album and its tracks to albums dictionary
+			albums[album] = tracks
+
```

Comment:
Your loop looks really good. Logic seems to all make sense. I would definitely add some sort of check-handle case here in your root for loop. You have this initial ' if os.path.isdir(album_path): ' statement, but there is no else-statement that handles the case of the statement not being satisfied. 

Or rather, reverse the logic. Could write it as, 'if not os.path.isdir(album_path): ' and within this block, handle the case where the path is invalid or presents some error. And from here you could output, "Some error with this directory" etc. And in the case it passes the if statement, can continue on in the else portion. 

Overall looks great. Definitely needs the error checking and handling functionality added though.